### PR TITLE
chore: make tests quieter

### PR DESCRIPTION
**Pull request by hayd**
Don't mix http request logging in with the tests output.
Don't print that the file servers are starting unless -vv flag is passed.

Capture the output of run with run_output which returns stdout, stderr and exit_code.
Test against exit_code rather than relying on sys.exit.

cc @bartlomieju @ry

### CODE REVIEW

File: tools/http_server.py [hunk 1]
```diff
@@ -17,8 +17,17 @@
 ANOTHER_REDIRECT_PORT = 4547
 DOUBLE_REDIRECTS_PORT = 4548
 
+QUIET = '-vv' not in sys.argv and sys.argv.count('-v') < 2
```
Comment (bartlomieju) [inline tools/http_server.py]: Can `QUIET` be taken out from arg parser?

Comment (hayd) [reply tools/http_server.py]: I think not since it's imported by benchmark.py and maybe others, which have different args than test.py...

File: tools/benchmark.py [hunk 2]
```diff
@@ -212,7 +212,8 @@ def main(argv):
         print "Usage: tools/benchmark.py [build_dir]"
         sys.exit(1)
 
-    sha1 = run_output(["git", "rev-parse", "HEAD"]).strip()
+    sha1 = run_output(["git", "rev-parse", "HEAD"],
+                      exit_on_fail=True).out.strip()
```
Comment (ry) [inline tools/benchmark.py]: Why is this being changed?

File: tools/fmt_test.py [hunk 3]
```diff
@@ -26,12 +26,15 @@ def test_fmt(self):
...
+            result = run_output([
                 os.path.join(root_path, self.deno_exe), "fmt",
                 "badly_formatted.js"
             ],
-                cwd=d,
-                merge_env={"DENO_DIR": deno_dir})
+                                cwd=d,
+                                merge_env={"DENO_DIR": deno_dir},
+                                exit_on_fail=True,
+                                quiet=True)
```
Comment (ry) [inline tools/fmt_test.py]: yapf \o/

File: tools/util.py [hunk 4]
```diff
@@ -59,15 +60,41 @@ def run(args, quiet=False, cwd=None, env=None, merge_env=None):
...
+    except subprocess.CalledProcessError as e:
+        p.kill()
+        p.wait()
+        raise e
+    retcode = p.poll()
+    if retcode and exit_on_fail:
+        sys.exit(retcode)
+    # Ignore Windows CRLF (\r\n).
+    return CmdResult(
+        out.replace('\r\n', '\n'), err.replace('\r\n', '\n'), retcode)
```
Comment (ry) [inline tools/util.py]: I see, the return type of run_output has changed. What is the purpose of this change?

Comment (hayd) [reply tools/util.py]: To capture stderr.

File: tools/target_test.py [hunk 5]
```diff
@@ -35,26 +35,31 @@ def test_core_http_benchmark(self):
...
         cmd = [self.deno_exe, "run", "tests/exec_path.ts"]
-        output = run_output(cmd)
-        assert self.deno_exe in output.strip()
+        result = run_output(cmd, quiet=True)
+        assert self.deno_exe in result.out.strip()
+        self.assertEqual(result.code, 0)
 
 
-if __name__ == "main":
+if __name__ == "__main__":
```
Comment (ry) [inline tools/target_test.py]: oops

File: tools/test_format.py [hunk 6]
```diff
@@ -9,11 +9,11 @@
 
 def main():
     util.run([sys.executable, "tools/format.py"])
-    output = util.run_output(
+    result = util.run_output(
         ["git", "status", "-uno", "--porcelain", "--ignore-submodules"])
```
Comment (ry) [inline tools/test_format.py]: exit_on_fail ?

**Comment by hayd** [comment]
In this branch you need to use -vv to show the http_server requests.
Sure, we could do -v as the default, the single dot output is a little overly concise...

**Comment by ry** [comment]
> In this branch you need to use -vv to show the http_server requests.

Cool

**Comment by bartlomieju** [comment]
> Sure, we could do -v as the default, the single dot output is a little overly concise...

+1 to that, -v should be `verbosity=2` in unittest
